tests/integration/override_executor_specific_params/test_specific_params.py

```diff
@@ -3,20 +3,48 @@
 from jina import Flow, DocumentArray, Document, Executor, requests
 from tests import validate_callback
 
+ORIGINAL_PARAMS = {'param1': 50, 'param2': 60, 'exec_name': {'param1': 'changed'}}
+OVERRIDEN_POD1_PARAMS = {
+    'param1': 'changed',
+    'param2': 60,
+    'exec_name': {'param1': 'changed'},
+}
+OVERRIDEN_POD2_PARAMS = {
+    'param1': 50,
+    'param2': 'change_in_pod',
+    'exec_name': {'param1': 'changed'},
+}
 
-class DummyOverrideParams(Executor):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
 
+class DummyOverrideParams(Executor):
     @requests()
     def bar(self, docs: 'DocumentArray', parameters: Dict, *args, **kwargs):
         for doc in docs:
             doc.tags = parameters
 
 
+class DummyAssertNotOverrideBetweenPodsParams(Executor):
+    @requests()
+    def bar(self, parameters: Dict, *args, **kwargs):
+        assert parameters == ORIGINAL_PARAMS
+        parameters['param2'] = 'change_in_pod'
+
+
+class DummyAssertIfParamsCanBeChangedInsidePods(Executor):
+    @requests()
+    def bar(self, parameters: Dict, *args, **kwargs):
+        # this test is not sure it is intended, but good way of documenting
+        assert parameters == ORIGINAL_PARAMS
+
+
 def test_override_params(mocker):
-    f = Flow(return_results=True).add(
-        uses={'jtype': 'DummyOverrideParams', 'metas': {'name': 'exec_name'}},
+    f = (
+        Flow(return_results=True)
+        .add(
+            uses={'jtype': 'DummyOverrideParams', 'metas': {'name': 'exec_name'}},
+        )
+        .add(uses=DummyAssertNotOverrideBetweenPodsParams)
+        .add(uses=DummyAssertIfParamsCanBeChangedInsidePods)
```

Comment:
Just to make things simpler, this could be just
```
        .add(uses=DummyAssertNotOverrideBetweenPodsParams)
        .add(uses=DummyAssertNotOverrideBetweenPodsParams)
```

Reply:
I want to make explicit that they are testing different things